lib/board.rb

```diff
@@ -61,7 +61,14 @@ def valid_coordinates_suite(ship)
 
   def valid_placement?(ship, coordinates)
     @coordinates = coordinates
-    valid_coordinates_suite(ship) && ship_overlap(ship, coordinates)
+    @ship = ship
+    valid_coordinates_suite(ship) && ship_overlap(ship, @coordinates)
+  end
+
+  def validated_placement
+    if (valid_coordinates_suite(@ship) && ship_overlap(@ship, @coordinates)) == true
```

Comment:
I see the call of @ship. I think we need to human speak this out because I'm not understanding it. Thanks in advance for your willingness to share knowledge!

Reply:
OMG... I get it now. It's so you have an option to place ship inside board without us needing to write that explicit code into the player class. AMAZING refactor!!! Love it and I see the connection :)